presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestJoinQueries.java

```diff
@@ -33,11 +39,62 @@
 import static com.facebook.presto.tests.QueryTemplate.queryTemplate;
 import static com.google.common.collect.Iterables.getOnlyElement;
 import static java.lang.String.format;
+import static org.testng.Assert.assertNotEquals;
 import static org.testng.Assert.assertTrue;
 
 public abstract class AbstractTestJoinQueries
         extends AbstractTestQueryFramework
 {
+    @Test
+    public void testShuffledStatsWithInnerJoin()
+    {
+        // NOTE: only test shuffled stats with distributed query runner.
+        if (!(getQueryRunner() instanceof DistributedQueryRunner)) {
+            return;
+        }
+        DistributedQueryRunner queryRunner = (DistributedQueryRunner) getQueryRunner();
+
+        // Get the number of rows in orders table for query stats verification below.
+        Long numOfOrdersRows = getTableRowCount("orders");
+        // Get the number of rows in lineitem table for query stats verification below.
+        Long numOfLineitemRows = getTableRowCount("lineitem");
+
+        String query = "SELECT a.orderkey, a.orderstatus, b.linenumber FROM orders a JOIN lineitem b ON a.orderkey = b.orderkey";
+        // Set session property to enforce a hash partitioned join.
+        Session sessionWithPartitionedJoin = Session.builder(getSession())
+                .setSystemProperty(JOIN_DISTRIBUTION_TYPE, PARTITIONED.name())
+                .setSystemProperty(JOIN_REORDERING_STRATEGY, NONE.name())
+                .build();
+        QueryId partitionQueryId = queryRunner.executeWithQueryId(sessionWithPartitionedJoin, query).getQueryId();
+        QueryInfo partitionJoinQueryInfo = queryRunner.getQueryInfo(partitionQueryId);
+        long expectedRawInputRows = numOfOrdersRows + numOfLineitemRows;
```

Comment:
consistency: expectedRawInputRows vs. numOfOrdersRows

Consider renaming numOfOrdersRows and numOfLineitemRows to ordersRows and lineitemRows.